### ISSUE TITLE
php-fpm not working as expected

### ISSUE THREAD

**Issue by justin-oh** [laravel/homestead#1617]
## Versions
  - Vagrant: 2.2.14 (tried upgrading from 2.2.7)
  - Provider: Virtualbox 6.1.4
  - Homestead: v10.17.0, v11.4.0 (shown after running `vagrant ssh`)

## Host operating system
macOS Catalina 10.15.7

## Homestead.yaml
```yaml
ip: 192.168.10.10
memory: 2048
cpus: 2
provider: virtualbox
authorize: ~/.ssh/id_rsa.pub
keys:
    - ~/.ssh/id_rsa
folders:
    - { map: ~/ohdev/phpmyadmin, to: /home/vagrant/phpmyadmin }
    - { map: ~/ohdev/sites/cicv5.ohdev/cicv5, to: /home/vagrant/cicv5 }
    - { map: ~/ohdev/sites/cjem-v5.ohdev/cjem-v5, to: /home/vagrant/cjem-v5 }
    - { map: ~/ohdev/sites/cypresshills-v5.ohdev/cypresshills-v5, to: /home/vagrant/cypresshills-v5 }
    - { map: ~/ohdev/sites/edr-agfood-v5.ohdev/edr-agfood-v5, to: /home/vagrant/edr-agfood-v5 }
    - { map: ~/ohdev/sites/fieldsmart-v5.ohdev/fieldsmart-v5, to: /home/vagrant/fieldsmart-v5 }
    - { map: ~/ohdev/sites/hvac-v5.ohdev/hvac-v5, to: /home/vagrant/hvac-v5 }
    - { map: ~/ohdev/sites/midwesteng-v5.ohdev/midwesteng-v5, to: /home/vagrant/midwesteng-v5 }
    - { map: ~/ohdev/sites/ohcmsv5.ohdev/ohcmsv5, to: /home/vagrant/ohcmsv5 }
    - { map: ~/ohdev/sites/phoenix-v5.ohdev/phoenix-v5, to: /home/vagrant/phoenix-v5 }
    - { map: ~/ohdev/sites/prescreener.ohdev/prescreener, to: /home/vagrant/prescreener }
    - { map: ~/ohdev/sites/rsm-v5.ohdev/rsm-v5, to: /home/vagrant/rsm-v5 }
    - { map: ~/ohdev/sites/sachm-v5.ohdev/sachm-v5, to: /home/vagrant/sachm-v5 }
    - { map: ~/ohdev/sites/sgeu-v5.ohdev/sgeu-v5, to: /home/vagrant/sgeu-v5 }
sites:
    - { map: cicv5.ohdev, to: /home/vagrant/cicv5/public_html, type: apache, php: '7.2' }
    - { map: cjem-v5.ohdev, to: /home/vagrant/cjem-v5/public_html, type: apache, php: '7.2' }
    - { map: cypresshills-v5.ohdev, to: /home/vagrant/cypresshills-v5/public_html, type: apache, php: '7.2' }
    - { map: edr-agfood-v5.ohdev, to: /home/vagrant/edr-agfood-v5/public_html, type: apache, php: '7.2' }
    - { map: fieldsmart-v5.ohdev, to: /home/vagrant/fieldsmart-v5/public_html, type: apache, php: '7.2' }
    - { map: hvac-v5.ohdev, to: /home/vagrant/hvac-v5/public_html, type: apache, php: '7.2' }
    - { map: midwesteng-v5.ohdev, to: /home/vagrant/midwesteng-v5/public_html, type: apache, php: '7.2' }
    - { map: ohcmsv5.ohdev, to: /home/vagrant/ohcmsv5/public_html, type: apache, php: '7.2' }
    - { map: phoenix-v5.ohdev, to: /home/vagrant/phoenix-v5/public_html, type: apache, php: '7.2' }
    - { map: phpmyadmin.ohdev, to: /home/vagrant/phpmyadmin, type: apache, php: '7.4' }
    - { map: prescreener.ohdev, to: /home/vagrant/prescreener/public, type: apache, php: '7.4' }
    - { map: rsm-v5.ohdev, to: /home/vagrant/rsm-v5/public_html, type: apache, php: '7.2' }
    - { map: sachm-v5.ohdev, to: /home/vagrant/sachm-v5/public_html, type: apache, php: '7.2' }
    - { map: sgeu-v5.ohdev, to: /home/vagrant/sgeu-v5/public_html, type: apache, php: '7.2' }
databases:
    - cicv5
    - cjem-v5
    - cypresshills-v5
    - edr-agfood-v5
    - fieldsmart-v5
    - hvac-v5
    - midwesteng-v5
    - ohcmsv5
    - phoenix-v5
    - prescreener
    - rsm-v5
    - sachm-v5
    - sgeu-v5
features:
    - { mariadb: false }
    - { ohmyzsh: false }
    - { webdriver: false }
name: ohdev
hostname: ohdev
```

## after.sh additions

```yaml
sudo apt-get install -y imagemagick php-imagick

sudo apt install -y ruby

sudo npm install -g sass

sudo update-alternatives --set php /usr/bin/php7.2
sudo update-alternatives --set phpize /usr/bin/phpize7.2
sudo update-alternatives --set php-config /usr/bin/php-config7.2

php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"

HASH1="$(curl -sb -H https://composer.github.io/installer.sig)"

HASH2="$(php -r "echo hash_file('SHA384', 'composer-setup.php');")"

if [ "$HASH1" = "$HASH2" ]
then
    sudo php composer-setup.php --install-dir=/usr/local/bin --filename=composer
    sudo chown vagrant:vagrant /usr/local/bin/composer
fi

php -r "unlink('composer-setup.php');"
```

I'm making sure the command-line version of PHP is 7.2 and that Composer is version 2. (Other people run this setup, and some of their Composers were coming in as 1.x for whatever reason.)

## Vagrant destroy & up output
https://gist.github.com/justin-oh/dafd80299dff72dfa60579ec536be56d

**Note:** this particular `vagrant up` output ends with an error. My workaround for that is to `vagrant ssh`, create the `.composer` directory manually, then `vagrant reload --provision`. I would like to know a fix for this if possible, but the FPM issue is more important.

## Expected behavior
Basically, all the "v5" sites are running off the same base code. I've been adding these sites in the same manner over the past weeks and they have all been working fine, running on PHP 7.2.

## Actual behavior
All sites are suddenly running off of PHP 7.4 and seemingly ignoring the FPM configuration. I can remedy this by removing PHP 7.4, but that's not a viable solution.

## Steps to reproduce
I have no clue how or why this is happening. It was working just fine yesterday until I added another one of those v5 sites and reloaded the provision. Is there a limit to how many sites can run on fpm? Is it getting log-jammed (for a lack of a better term)?

## References
N/A

**Comment by svpernova09** [comment]
Are you able to reproduce on the current `release` branch with the newer base box? 

You can remove the Composer stuff in your `after.sh`, Composer is already installed. You're getting an error during vagrant up because it's composer 2 without the newer `~/.config` path. (Newer base box fixes this)

**Comment by justin-oh** [comment]
> Are you able to reproduce on the current `release` branch with the newer base box?
> 
> You can remove the Composer stuff in your `after.sh`, Composer is already installed. You're getting an error during vagrant up because it's composer 2 without the newer `~/.config` path. (Newer base box fixes this)

I mentioned on some other people's machines Composer was being installed with version 1.x which is why I added the extra lines make sure it installs Composer 2.x.

I could try the updated base box; I will need an afternoon. Are you referring to `laravel/homestead`? I did try `vagrant box update` or `vagrant update box` (whatever it was).

Regardless, I don't see how that caused my issue out of nowhere. Basically all I did was update Homestead.yaml to tell it about a new site and `vagrant reload --provision`. Nothing else should have changed. Any version changes I made after the fact was in an effort to remedy the situation.

**Comment by svpernova09** [comment]
> I mentioned on some other people's machines Composer was being installed with version 1.x which is why I added the extra lines make sure it installs Composer 2.x.

This is an indicator they're using an older base box with a newer repo version. We've only recently started tightly coupling these to help reduce these occurrences.

> I could try the updated base box; I will need an afternoon. Are you referring to `laravel/homestead`? I did try `vagrant box update` or `vagrant update box` (whatever it was).
> 
> Regardless, I don't see how that caused my issue out of nowhere. Basically all I did was update Homestead.yaml to tell it about a new site and `vagrant reload --provision`. Nothing else should have changed. Any version changes I made after the fact was in an effort to remedy the situation.

100% get the time issue. I would say the fastest way to fix your issue is to sudo rm the virtualhost files for the broken/not working sites and then try to provision again.

These boxes are snapshots in time. When you start and stop them weird things can happen over time. Vagrant machines aren't designed to be long-running hosts. This is why you see new base boxes built every 2-3 months or sooner. The only version of Homestead I'm able to support here is the `release` branch, which is always the most recent stable release.

I think the issue you're running into is because the apache scripts aren't idempotent, so if you run them once, and run them again. You're not going to be guaranteed the same results each consecutive run. Now for the most part this works. which is why it seems to have started when you "reloaded the provision". There are dozens of things that can go wrong especially since the version you're using doesn't have Apache already baked into the base box. It should be a faster/smoother experience now.
